htmresearch/frameworks/nlp/imbu.py

```diff
@@ -0,0 +1,203 @@
+import argparse
+import os
+
+from enum import Enum
+
+from htmresearch.encoders import EncoderTypes
+from htmresearch.frameworks.nlp.classification_model import ClassificationModel
+from htmresearch.frameworks.nlp.classify_fingerprint import (
+  ClassificationModelFingerprint)
+from htmresearch.frameworks.nlp.classify_htm import ClassificationModelHTM
+from htmresearch.frameworks.nlp.classify_keywords import (
+  ClassificationModelKeywords)
+from htmresearch.frameworks.nlp.classify_windows import (
+  ClassificationModelWindows)
+from htmresearch.support.csv_helper import readCSV
+from htmresearch.support.text_preprocess import TextPreprocess
+
+
+
+class ImbuModelTypes(Enum):
+  CioWordFingerprint = ClassificationModelFingerprint
+  CioDocumentFingerprint = ClassificationModelFingerprint
+  CioWindows = ClassificationModelWindows
+  Keywords = ClassificationModelKeywords
+  HTMNetwork = ClassificationModelHTM
+
+
+
+class ModelSimilarityMetrics(Enum):
+  pctOverlapOfInput = "pctOverlapOfInput"
+
+
+
+class ImbuError(Exception):
+  pass
+
+
+
+class ImbuUnableToLoadModelError(ImbuError):
+  def __init__(self, exc):
+    self.exc = exc
+
+
+
+class ImbuModels(object):
```

Comment:
Is the intent that this class would be used by the Imbu application? Not clear to me these methods match the Imbu UI and user workflow.  Seems to be a simple wrapper around the ClassificationModel class that doesn't do too much. Am I missing something?